File: bagging/src/main/java/org/example/bagging/entity/ParsianLoan.java
package org.example.bagging.entity;

import jakarta.persistence.*;
import lombok.Data;

import java.math.BigDecimal;
import java.time.LocalDate;
import java.time.LocalDateTime;

@Data
@Entity
@Table(name = "PARSIAN_LOAN")
public class ParsianLoan {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "ID")
    private Long id;

    @Column(name = "INSERT_SYSDATE", nullable = false)
    private LocalDateTime insertSysdate;

    @Column(name = "branch_code", nullable = false)
    private Integer branchCode;

    @Column(name = "branchname", columnDefinition = "VARCHAR(100) CHARACTER SET utf8mb4 COLLATE utf8mb4_unicode_ci")
    private String branchName;

    @Column(name = "loan_file_numberr")
    private Long clientId;

    @Column(name = "CLIENT_ID")
    private Integer clientIdNumber;

    @Column(name = "sit_flag")
    private Character sitFlag;

    @Column(name = "interest_rate", precision = 19, scale = 2)
    private BigDecimal interestRate;

    @Column(name = "total_repayment_up_to_now", precision = 28, scale = 8)
    private BigDecimal totalRepaymentUpToNow;

    @Column(name = "COMMISSION_AMOUNT_REMAIN", precision = 28, scale = 8)
    private BigDecimal commissionAmountRemain;

    @Column(name = "CHARGE", precision = 28, scale = 8)
    private BigDecimal charge;

    @Column(name = "DISCOUNT", precision = 28, scale = 8)
    private BigDecimal discount;

    @Column(name = "ADVANCE_PAY_TO_TOTAL_CASH", precision = 28, scale = 8)
    private BigDecimal advancePayToTotalCash;

    @Column(name = "ADVANCE_PAY_TO_REMAIN_NON_CASH", precision = 28, scale = 8)
    private BigDecimal advancePayToRemainNonCash;

    @Column(name = "is_installment")
    private Character isInstallment;

    @Column(name = "interest_sum", precision = 28, scale = 8)
    private BigDecimal interestSum;

    @Column(name = "INSTALLMENT_NUMBER_REMAIN")
    private Integer installmentNumberRemain;

    @Column(name = "RECEIVABLE_INSTALLMENT_NUMBER")
    private Integer receivableInstallmentNumber;

    @Column(name = "FIRST_PASSED")
    private LocalDate firstPassed;

    @Column(name = "TOTAL_PAYMENT_UP_TO_NOW", precision = 28, scale = 8)
    private BigDecimal totalPaymentUpToNow;

    @Column(name = "FINALIZED_LOAN_AMOUNT", precision = 28, scale = 8)
    private BigDecimal finalizedLoanAmount;

    @Column(name = "PENALTY", precision = 28, scale = 8)
    private BigDecimal penalty;

    @Column(name = "FIRST_PAYMENT_DATE_IN_DU")
    private LocalDate firstPaymentDateInDu;

    @Column(name = "PRINCIPAL_SUM", precision = 28, scale = 8)
    private BigDecimal principalSum;

    @Column(name = "ADVANCE_PAY", precision = 28, scale = 8)
    private BigDecimal advancePay;

    @Column(name = "SIT_DURATION")
    private Integer sitDuration;

    @Column(name = "sit_duration_day")
    private Integer sitDurationDay;

    @Column(name = "sit_distribute_phases")
    private Integer sitDistributePhases;

    @Column(name = "sit_fast_receive_percent")
    private Float sitFastReceivePercent;

    @Column(name = "frequency")
    private Integer frequency;

    @Column(name = "CUSTOMER_OBLIGATION_AMOUNT", precision = 28, scale = 8)
    private BigDecimal customerObligationAmount;

    @Column(name = "CUSTOMER_SHARE_CASH_AMOUNT", precision = 28, scale = 8)
    private BigDecimal customerShareCashAmount;

    @Column(name = "CUSTOMER_SHARE_NON_CASH_AMOUNT", precision = 28, scale = 8)
    private BigDecimal customerShareNonCashAmount;

    @Column(name = "BANK_SHARE_CASH_AMOUNT", precision = 28, scale = 8)
    private BigDecimal bankShareCashAmount;

    @Column(name = "BANK_SHARE_NON_CASH_AMOUNT", precision = 28, scale = 8)
    private BigDecimal bankShareNonCashAmount;

    @Column(name = "First_OverDue")
    private LocalDate firstOverDue;

    @Column(name = "LOAN_DURATION_DAY")
    private Integer loanDurationDay;

    @Column(name = "Loan_File_Number")
    private Long loanFileNumber;

    @Column(name = "CreateDate")
    private LocalDate createDate;

    @Column(name = "LONG_TITLE", columnDefinition = "VARCHAR(255) CHARACTER SET utf8mb4 COLLATE utf8mb4_unicode_ci")
    private String longTitle;

    @Column(name = "Status", columnDefinition = "VARCHAR(255) CHARACTER SET utf8mb4 COLLATE utf8mb4_unicode_ci")
    private String status;

    @Column(name = "contract", columnDefinition = "VARCHAR(255) CHARACTER SET utf8mb4 COLLATE utf8mb4_unicode_ci")
    private String contract;

    @Column(name = "APPROVAL_AMOUNT", precision = 28, scale = 8)
    private BigDecimal approvalAmount;

    @Column(name = "TITLE", columnDefinition = "VARCHAR(255) CHARACTER SET utf8mb4 COLLATE utf8mb4_unicode_ci")
    private String title;

    @Column(name = "inc_COMMISSION_AMOUNT", precision = 28, scale = 8)
    private BigDecimal incCommissionAmount;

    @Column(name = "INTEREST_AMOUNT", precision = 28, scale = 8)
    private BigDecimal interestAmount;

    @Column(name = "OBLIGATION_PENALTY", precision = 28, scale = 8)
    private BigDecimal obligationPenalty;

    @Column(name = "PASSED_DATE")
    private LocalDate passedDate;

    @Column(name = "PENALTY_INTEREST", precision = 28, scale = 8)
    private BigDecimal penaltyInterest;

    @Column(name = "TO_DUE_DATE", precision = 28, scale = 8)
    private BigDecimal toDueDate;

    @Column(name = "TO_END_OF_MONTH", precision = 28, scale = 8)
    private BigDecimal toEndOfMonth;

    @Column(name = "DUE_DATE")
    private LocalDate dueDate;
}

File: PythonProject1/ParsianLoan.py
from sqlalchemy import (
    Column, Integer, BigInteger, String, Date, DateTime, Numeric, Float
)
from sqlalchemy.orm import declarative_base
from datetime import datetime

Base = declarative_base()

class ParsianLoan(Base):
    __tablename__ = "parsian_loan"

    id = Column("id", BigInteger, primary_key=True, autoincrement=True)
    insert_sysdate = Column("insert_sysdate", DateTime, nullable=False, default=datetime.utcnow)
    branch_code = Column("branch_code", Integer, nullable=False)
    branchname = Column("branchname", String(100), nullable=True)

    client_id = Column("client_id", Integer, nullable=True)
    loan_file_numberr = Column("loan_file_numberr", BigInteger, nullable=True)

    sit_flag = Column("sit_flag", String(1), nullable=True)
    interest_rate = Column("interest_rate", Numeric(19, 2), nullable=True)
    total_repayment_up_to_now = Column("total_repayment_up_to_now", Numeric(28, 8), nullable=True)
    commission_amount_remain = Column("commission_amount_remain", Numeric(28, 8), nullable=True)
    charge = Column("charge", Numeric(28, 8), nullable=True)
    discount = Column("discount", Numeric(28, 8), nullable=True)

    advance_pay_to_total_cash = Column("advance_pay_to_total_cash", Numeric(28, 8), nullable=True)
    advance_pay_to_remain_non_cash = Column("advance_pay_to_remain_non_cash", Numeric(28, 8), nullable=True)
    is_installment = Column("is_installment", String(1), nullable=True)
    interest_sum = Column("interest_sum", Numeric(28, 8), nullable=True)

    installment_number_remain = Column("installment_number_remain", Integer, nullable=True)
    receivable_installment_number = Column("receivable_installment_number", Integer, nullable=True)
    first_passed = Column("first_passed", Date, nullable=True)
    total_payment_up_to_now = Column("total_payment_up_to_now", Numeric(28, 8), nullable=True)
    finalized_loan_amount = Column("finalized_loan_amount", Numeric(28, 8), nullable=True)
    penalty = Column("penalty", Numeric(28, 8), nullable=True)

    first_payment_date_in_du = Column("first_payment_date_in_du", Date, nullable=True)
    principal_sum = Column("principal_sum", Numeric(28, 8), nullable=True)
    advance_pay = Column("advance_pay", Numeric(28, 8), nullable=True)

    sit_duration = Column("sit_duration", Integer, nullable=True)
    sit_duration_day = Column("sit_duration_day", Integer, nullable=True)
    sit_distribute_phases = Column("sit_distribute_phases", Integer, nullable=True)
    sit_fast_receive_percent = Column("sit_fast_receive_percent", Float, nullable=True)

    frequency = Column("frequency", Integer, nullable=True)
    customer_obligation_amount = Column("customer_obligation_amount", Numeric(28, 8), nullable=True)
    customer_share_cash_amount = Column("customer_share_cash_amount", Numeric(28, 8), nullable=True)
    customer_share_non_cash_amount = Column("customer_share_non_cash_amount", Numeric(28, 8), nullable=True)

    bank_share_cash_amount = Column("bank_share_cash_amount", Numeric(28, 8), nullable=True)
    bank_share_non_cash_amount = Column("bank_share_non_cash_amount", Numeric(28, 8), nullable=True)
    first_over_due = Column("first_over_due", Date, nullable=True)  # اصلاح شده
    loan_duration_day = Column("loan_duration_day", Integer, nullable=True)

    loan_file_number = Column("loan_file_number", BigInteger, nullable=True)
    create_date = Column("create_date", Date, nullable=True)
    long_title = Column("long_title", String(255), nullable=True)
    status = Column("status", String(255), nullable=True)
    contract = Column("contract", String(255), nullable=True)

    approval_amount = Column("approval_amount", Numeric(28, 8), nullable=True)
    title = Column("title", String(255), nullable=True)
    inc_commission_amount = Column("inc_commission_amount", Numeric(28, 8), nullable=True)
    interest_amount = Column("interest_amount", Numeric(28, 8), nullable=True)

    obligation_penalty = Column("obligation_penalty", Numeric(28, 8), nullable=True)
    passed_date = Column("passed_date", Date, nullable=True)
    penalty_interest = Column("penalty_interest", Numeric(28, 8), nullable=True)

    to_due_date = Column("to_due_date", Numeric(28, 8), nullable=True)
    to_end_of_month = Column("to_end_of_month", Numeric(28, 8), nullable=True)
    due_date = Column("due_date", Date, nullable=True)

    def __repr__(self):
        return f"<ParsianLoan(id={self.id}, branch_code={self.branch_code}, client_id={self.client_id})>"
